diff --git a/fabric/v4/api_routing_protocols.go b/fabric/v4/api_routing_protocols.go
index 7f9552e..75c325e 100644
--- a/fabric/v4/api_routing_protocols.go
+++ b/fabric/v4/api_routing_protocols.go
@@ -12,6 +12,7 @@ package v4
 import (
 	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
 	"io"
 	"io/ioutil"
@@ -29,6 +30,36 @@ var (
 
 type RoutingProtocolsApiService service
 
+func (v *RoutingProtocolData) UnmarshalJSON(data []byte) error {
+	strData := string(data)
+	if strData == "null" || strData == `""` {
+		return nil
+	}
+
+	var (
+		d RoutingProtocolDirectData
+		b RoutingProtocolBgpData
+	)
+
+	if strings.Contains(strData, "DIRECT") {
+		if err := json.Unmarshal(data, &d); err != nil {
+			return err
+		}
+		v.Type_ = "DIRECT"
+		v.RoutingProtocolDirectData = d
+	}
+
+	if strings.Contains(strData, "BGP") {
+		if err := json.Unmarshal(data, &b); err != nil {
+			return err
+		}
+		v.Type_ = "BGP"
+		v.RoutingProtocolBgpData = b
+	}
+
+	return nil
+}
+
 /*
 RoutingProtocolsApiService Create Protocol
 This API provides capability to create Routing Protocol for connections
